test/controllers/CheckYourAnswersControllerSpec.scala

```diff
@@ -17,37 +17,93 @@
+import play.api.i18n.Messages
 
 class CheckYourAnswersControllerSpec extends SpecBase with SummaryListFluency {
+  val checkYourAnswersView: CheckYourAnswersView         = app.injector.instanceOf[CheckYourAnswersView]
+  val mockCheckYourAnswersHelper: CheckYourAnswersHelper = mock[CheckYourAnswersHelper]
 
-  private val checkYourAnswersView: CheckYourAnswersView = app.injector.instanceOf[CheckYourAnswersView]
+  val checkYourAnswersController = new CheckYourAnswersController(
+    stubMessagesControllerComponents(),
+    new FakeAuthoriseAction(defaultBodyParser),
+    checkYourAnswersView,
+    sessionRequest,
+    mockCheckYourAnswersHelper
+  )
 
-  "Check Your Answers Controller" - {
+  "CheckYourAnswersController" - {
 
-    "must return OK and the correct view for a onPageLoad with user answers set" in {
-      val checkYourAnswersControllerWithData = new CheckYourAnswersController(
-        messagesApi,
-        new FakeAuthoriseAction(defaultBodyParser),
-        new FakeSessionRequestAction(emptyUserAnswers),
-        messageComponentControllers,
-        checkYourAnswersView
-      )
+    "must return OK and the correct view for a GET" in {
 
-      val result = checkYourAnswersControllerWithData.onPageLoad()(fakeRequest)
+      val summaryList = List(
+        SummaryListRow(
+          Key(HtmlContent("UKIMS number")),
+          Value(HtmlContent("11")),
+          "",
+          Some(
+            Actions(
+              "",
+              List(ActionItem("/trader-goods-profiles/ukims-number/check", HtmlContent("Change"), None, "", Map()))
+            )
+          )
+        ),
+        SummaryListRow(
+          Key(HtmlContent("NIRMS registered"), ""),
+          Value(HtmlContent("No"), ""),
+          "",
+          Some(
+            Actions(
+              "",
+              List(ActionItem("/trader-goods-profiles/nirms-question/check", HtmlContent("Change"), None, "", Map()))
+            )
+          )
+        ),
+        SummaryListRow(
+          Key(HtmlContent("NIPHL registered"), ""),
+          Value(HtmlContent("No"), ""),
+          "",
+          Some(
+            Actions(
+              "",
+              List(ActionItem("/trader-goods-profiles/niphl-question/check", HtmlContent("Change"), None, "", Map()))
+            )
+          )
+        )
+      )
 
-      val expectedList = SummaryListViewModel(
-        rows = Seq.empty
+      val list = SummaryListViewModel(
+        rows = summaryList
       )
 
+      when(mockCheckYourAnswersHelper.createSummaryList(any[TraderGoodsProfile])(any())) thenReturn summaryList
+
+      val result = checkYourAnswersController.onPageLoad()(fakeRequest)
+
       status(result) mustEqual OK
 
-      contentAsString(result) mustEqual checkYourAnswersView(expectedList)(fakeRequest, messages).toString
+      contentAsString(result) mustEqual checkYourAnswersView(list)(fakeRequest, stubMessages()).toString()
 
     }
 
+    "must redirect on Submit" in {
```

Comment:
can we add a test for the unhappy path/s

Reply:
I'm not sure there is an unhappy path atm because it just directs to the dummy controller regardless on submit